### ISSUE TITLE
Example of `mutate` from ibis doctests fails to compile in ibis-substrait

### ISSUE THREAD

**Issue by rtpsw** [ibis-project/ibis-substrait#132]
To reproduce the issue for the [doctest example](https://github.com/ibis-project/ibis/blob/064d53e19e294aff80c981b9846220ce47ca6950/ibis/expr/types/relations.py#L539-L544):
```
>>> import ibis
>>> table = ibis.table([('foo', 'double'), ('bar', 'double')], name='t')
>>> expr = table.mutate(qux=table.foo + table.bar, baz=5)
>>> from ibis_substrait.compiler.core import SubstraitCompiler
>>> compiler = SubstraitCompiler()
>>> compiler.compile(expr)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/user1/tscontract/github/rtpsw/ibis-substrait/ibis_substrait/compiler/core.py", line 84, in compile
    input=translate(expr.op(), expr, self),
  File "/usr/lib/python3.8/functools.py", line 875, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/mnt/user1/tscontract/github/rtpsw/ibis-substrait/ibis_substrait/compiler/translate.py", line 664, in selection
    project=stalg.ProjectRel(
TypeError: Parameter to MergeFrom() must be instance of same class: expected substrait.Expression got substrait.Rel.
```
The error says that protobuf is asked to merge the [expressions of the selections](https://github.com/ibis-project/ibis-substrait/blob/5d3d26414c21fa449bf2ab6bf1c87ce3920c51ba/ibis_substrait/compiler/translate.py#L654-L662), but they are not all of the same type as required; in particular, the first (table) selection translates to a `Rel` message while the others translate to a `Expression` message.

**Comment by rtpsw** [comment]
cc @cpcloud, @icexelloss

**Comment by icexelloss** [comment]
The issue might be that ibis's selection structure doesn't match substrait's selection structure

For mutate ibis has a TableExpr (original columns) and a list of ColumExpr (new columns). I *think* what we need to do here is to change the translation of Selection to destructure the TableExpr into a list of substrait Expression. 

Curious to hear @cpcloud 's thought

**Comment by cpcloud** [comment]
Closed by #185.
